packages/diagram/lib/diagram.dart

```diff
@@ -0,0 +1,442 @@
+  ///
+  /// The file will be written in whatever format the suffix on the file
+  /// indicates. Valid suffixes are ".jpg", ".jpeg", ".png", ".gif", and ".tga".
+  Future<File> drawDiagramToFile(
+    File outputFile, {
+    ui.ImageByteFormat format: ui.ImageByteFormat.png,
+  }) async {
+    assert(outputFile != null);
+    if (!outputFile.isAbsolute && outputDirectory != null) {
+      // If output path is relative, make it relative to the output directory.
+      outputFile = new File(path.join(outputDirectory.absolute.path, outputFile.path));
+    }
+    ui.Image captured = await drawDiagramToImage();
+    ByteData encoded = await captured.toByteData(format: format);
+    List<int> bytes = encoded.buffer.asUint8List().toList();
+    print('Writing ${bytes.length} bytes, ${captured.width}x${captured.height} '
+        '${_byteFormatToString(format)}, to ${outputFile.absolute.path}.');
+    await outputFile.writeAsBytes(bytes);
+    return outputFile;
+  }
+
+  /// Draws multiple frames of the widget generated by [builder] into multiple
+  /// images and returns them.
+  ///
+  /// The animation will start at timestamp [start], end at [end], with the time
+  /// between frames specified by [frameDuration].  For instance, to generate
+  /// 11 frames of an animation that is 1 second long, set [start] to
+  /// [Duration.zero] (the default), set [end] to `const Duration(seconds: 1)`,
+  /// and set [frameDuration] to `const Duration(milliseconds: 100)`. The
+  /// animation includes the frames at [start] and [end].
+  ///
+  /// None of the parameters may be null. The [start] parameter must be less
+  /// than or equal to the [end] parameter, and the [frameDuration] must be
+  /// less than or equal to the time between [start] and [end]. All durations
+  /// must be greater than [Duration.zero].
+  Future<List<ui.Image>> drawAnimatedDiagramToImages({
+    Duration start: Duration.zero,
+    @required Duration end,
+    @required Duration frameDuration,
+  }) async {
+    assert(end != null);
+    assert(start != null);
+    assert(frameDuration != null);
+    assert(end >= start);
+    assert(frameDuration <= (end - start));
+    assert(frameDuration > Duration.zero);
+    assert(end > Duration.zero);
+    assert(start >= Duration.zero);
+
+    Duration now = start;
+    List<ui.Image> outputImages = <ui.Image>[];
+    while (now <= end) {
+      ui.Image captured = await drawDiagramToImage();
+      print('Generated frame for $now, ${captured.width}x${captured.height}');
+      outputImages.add(captured);
+      advanceTime(frameDuration);
+      now += frameDuration;
+    }
+    return outputImages;
+  }
+
+  /// Draws multiple frames of the widget generated by [builder] into multiple
+  /// files.
+  ///
+  /// The filenames are determined by [frameFilenameGenerator].
+  ///
+  /// The files will be written in whatever format the suffix on the file
+  /// indicates. Valid suffixes are ".jpg", ".jpeg", ".png", ".gif", and ".tga".
```

Comment:
I'm sure glad we still support Truevision TGA in 2018.

Reply:
LOL.  Yeah, me too.  It *is* one of the simplest formats ever to implement, however.